tests/TileDB.CSharp.Test/DeletesTest.cs

```diff
@@ -0,0 +1,928 @@
+            };
+
+            array.Open(QueryType.Read);
+            Assert.AreEqual(12UL, readQuery.ResultBufferElements()["a1"].Item1);
+            array.Close();
+            TestUtil.CompareBuffers(expectedData, readBuffers, duplicates);
+        }
+
+        // Params for all tests are (layout, legacy, duplicates)
+        [DataRow(LayoutType.Unordered, true, true)]
+        [DataRow(LayoutType.Unordered, true, false)]
+        [DataRow(LayoutType.Unordered, false, true)]
+        [DataRow(LayoutType.Unordered, false, false)]
+        [DataRow(LayoutType.RowMajor, true, true)]
+        [DataRow(LayoutType.RowMajor, true, false)]
+        [DataRow(LayoutType.RowMajor, false, true)]
+        [DataRow(LayoutType.RowMajor, false, false)]
+        [DataRow(LayoutType.GlobalOrder, true, true)]
+        [DataRow(LayoutType.GlobalOrder, true, false)]
+        [DataRow(LayoutType.GlobalOrder, false, true)]
+        [DataRow(LayoutType.GlobalOrder, false, false)]
+        [DataTestMethod]
+        public void MultipleCells(LayoutType layout, bool legacy, bool duplicates)
+        {
+            string arrayName = "deletes-multiple-cells";
+            var ctx = Context.GetDefault();
+            ctx.Config().Set("sm.query.sparse_global_order.reader", legacy ? "legacy" : "refactored");
+            ctx.Config().Set("sm.query.sparse_unordered_with_dups.reader", legacy ? "legacy" : "refactored");
+            InitTest(ctx, arrayName, layout, duplicates);
+            var array = new Array(ctx, arrayName);
+
+            DeleteFour(ctx, array, layout, duplicates);
+        }
+
+        [DataRow(LayoutType.Unordered, true, true)]
+        [DataRow(LayoutType.Unordered, true, false)]
+        [DataRow(LayoutType.Unordered, false, true)]
+        [DataRow(LayoutType.Unordered, false, false)]
+        [DataRow(LayoutType.RowMajor, true, true)]
+        [DataRow(LayoutType.RowMajor, true, false)]
+        [DataRow(LayoutType.RowMajor, false, true)]
+        [DataRow(LayoutType.RowMajor, false, false)]
+        [DataRow(LayoutType.GlobalOrder, true, true)]
+        [DataRow(LayoutType.GlobalOrder, true, false)]
+        [DataRow(LayoutType.GlobalOrder, false, true)]
+        [DataRow(LayoutType.GlobalOrder, false, false)]
+        [DataTestMethod]
+        public void SingleCell(LayoutType layout, bool legacy, bool duplicates)
+        {
+            string arrayName = "deletes-single-cell";
+            var ctx = Context.GetDefault();
+            ctx.Config().Set("sm.query.sparse_global_order.reader", legacy ? "legacy" : "refactored");
+            ctx.Config().Set("sm.query.sparse_unordered_with_dups.reader", legacy ? "legacy" : "refactored");
+            InitTest(ctx, arrayName, layout, duplicates);
+            var array = new Array(ctx, arrayName);
+
+            // Delete one cell
+            array.Open(QueryType.Delete);
+            var deleteQuery = new Query(ctx, array, QueryType.Delete);
```

Comment:
Can you add `using var` to all TileDB objects? This will dispose them at the end of the method without waiting for the garbage collector.